File: 1658 - Minimum Operations to Reduce X to Zero/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& A, int x) {
        int res = INT_MAX, l = 0, sum = 0, n = A.size();
        int target = accumulate(A.begin(), A.end(), 0)-x;
        if(target == 0) return n;
        for(int i = 0; i < n; i++){
            sum += A[i];
            while(sum>=target && l<=i){
                if(sum == target) res = min(res, n-(i-l+1));
                sum -= A[l++];
            }
        }
        return res==INT_MAX?-1:res;
    }
};

File: 1658 - Minimum Operations to Reduce X to Zero/solution.py
class Solution:
    def minOperations(self, A: List[int], x: int) -> int:
        res, s, l, n, target = float('inf'), 0, 0, len(A), sum(A)-x
        if target == 0:
            return n
        for i, v in enumerate(A):
            s += v
            while s >= target and l <= i:
                if s == target:
                    res = min(res, n-(i-l+1))
                s -= A[l]; l += 1
        return -1 if res == float('inf') else res
        